src/snek/driver.rs

```diff
@@ -1,78 +1,162 @@
 use crate::snek::game::Game;
 use crate::snek::terminal::Terminal;
-use std::io;
+use crossbeam_channel::{
+  self as crossbeam, bounded, TryRecvError, TrySendError,
+};
+use std::{io, thread, time};
 use termion::event::Key;
 use termion::input::TermRead;
 
 pub struct Driver {
   term: Terminal,
+  game: Game,
+  paused: bool,
 }
 
+pub type Result<T> = std::result::Result<T, String>;
+
 pub(crate) enum UserAction {
-  MoveNorth,
-  MoveSouth,
-  MoveEast,
-  MoveWest,
+  Move(Direction),
```

Comment:
very nice abstraction

Reply:
Made a couple things easier, glad you approve